packages/visx-zoom/src/Zoom.tsx

```diff
@@ -79,229 +80,267 @@ type ZoomState = {
-    this.startPoint = localPoint(event) || undefined;
-    this.startTranslate = { translateX, translateY };
-    this.setState({ isDragging: true });
-  };
+  const dragMove = useCallback(
+    (
+      event: React.MouseEvent | React.TouchEvent | MouseEvent | TouchEvent | React.PointerEvent,
+      options?: { offsetX?: number; offsetY?: number },
+    ) => {
+      if (!isDragging || !startPoint || !startTranslate) return;
+      const currentPoint = localPoint(event);
+      const dx = currentPoint ? -(startPoint.x - currentPoint.x) : -startPoint.x;
+      const dy = currentPoint ? -(startPoint.y - currentPoint.y) : -startPoint.y;
 
-  dragMove = (
-    event: React.MouseEvent | React.TouchEvent | MouseEvent | TouchEvent,
-    options?: { offsetX?: number; offsetY?: number },
-  ) => {
-    if (!this.state.isDragging || !this.startPoint || !this.startTranslate) return;
-    const currentPoint = localPoint(event);
-    const dx = currentPoint ? -(this.startPoint.x - currentPoint.x) : -this.startPoint.x;
-    const dy = currentPoint ? -(this.startPoint.y - currentPoint.y) : -this.startPoint.y;
+      let translateX = startTranslate.translateX + dx;
+      if (options?.offsetX) translateX += options?.offsetX;
+      let translateY = startTranslate.translateY + dy;
+      if (options?.offsetY) translateY += options?.offsetY;
+      setTranslate({
+        translateX,
+        translateY,
+      });
+    },
+    [isDragging, setTranslate, startPoint, startTranslate],
+  );
 
-    let translateX = this.startTranslate.translateX + dx;
-    if (options?.offsetX) translateX += options?.offsetX;
-    let translateY = this.startTranslate.translateY + dy;
-    if (options?.offsetY) translateY += options?.offsetY;
-    this.setTranslate({
-      translateX,
-      translateY,
-    });
-  };
+  const dragEnd = useCallback(() => {
+    setStartPoint(undefined);
+    setStartTranslate(undefined);
+    setIsDragging(false);
+  }, []);
 
-  dragEnd = () => {
-    this.startPoint = undefined;
-    this.startTranslate = undefined;
-    this.setState({ isDragging: false });
-  };
+  const handleWheel = useCallback(
+    (event: React.WheelEvent | WheelEvent) => {
+      event.preventDefault();
+      const point = localPoint(event) || undefined;
+      const { scaleX, scaleY } = wheelDelta!(event);
+      scale({ scaleX, scaleY, point });
+    },
+    [scale, wheelDelta],
+  );
 
-  handleWheel = (event: React.WheelEvent | WheelEvent) => {
-    const { passive, wheelDelta } = this.props;
-    if (!passive) event.preventDefault();
-    const point = localPoint(event) || undefined;
-    const { scaleX, scaleY } = wheelDelta!(event);
-    this.scale({ scaleX, scaleY, point });
-  };
+  const handlePinch: UserHandlers['onPinch'] = useCallback(
+    state => {
+      const {
+        origin: [ox, oy],
+      } = state;
+      if (containerRef.current) {
+        const { top, left } = containerRef.current.getBoundingClientRect();
```

Comment:
not sure how this call will be for perf or if it's really avoidable, but we can iterate in the future if need be.